awscli/s3transfer/copies.py

```diff
@@ -300,6 +323,52 @@ def _get_transfer_size(
             return total_transfer_size - (part_index * part_size)
         return part_size
 
+    def _check_destination_object_existence(self, client, call_args):
+        """
+        Check if destination object exists when IfNoneMatch is specified.
+
+        This method implements the ifNoneMatch header functionality by checking if the
```

Comment:
This statement is confusing. I'm not sure I understand why all of the implementation is based around `IfNoneMatch`? For example, we could base all functionality around head object pre-checks, and then any logic using `IfNoneMatch` would be altered to have that behavior?